tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas2_hemv_kk.hpp

```diff
@@ -0,0 +1,290 @@
+#define LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_HERMITIAN_MATVEC_HPP_
+
+namespace KokkosKernelsSTD {
+
+namespace hermitianmatvecimpl{
+
+template<class T> struct is_complex : std::false_type{};
+template<> struct is_complex<std::complex<float>> : std::true_type{};
+template<> struct is_complex<std::complex<double>> : std::true_type{};
+template<> struct is_complex<std::complex<long double>> : std::true_type{};
+
+template<class T> inline constexpr bool is_complex_v = is_complex<T>::value;
+
+} // end namespace hermitianmatvecimpl
+
+//
+// Overwriting hermitian matrix-vector product: y = Ax
+//
+// for now, specialize for default_accessor
+// https://github.com/kokkos/stdBLAS/issues/122
+//
+template<class ExeSpace,
+	 class ElementType_A,
+         std::experimental::extents<>::size_type numRows_A,
+         std::experimental::extents<>::size_type numCols_A,
+         class Layout_A,
+         class Triangle,
+         class ElementType_x,
+         std::experimental::extents<>::size_type ext_x,
+         class Layout_x,
+         class ElementType_y,
+         std::experimental::extents<>::size_type ext_y,
+         class Layout_y>
+requires (Layout_A::template mapping<std::experimental::extents<numRows_A, numCols_A>>::is_always_unique())
+void hermitian_matrix_vector_product(kokkos_exec<ExeSpace> kexe,
+				      std::experimental::mdspan<
+					ElementType_A,
+					std::experimental::extents<numRows_A, numCols_A>,
+					Layout_A,
+				        std::experimental::default_accessor<ElementType_A>
+				      > A,
+				      Triangle tr,
+				      std::experimental::mdspan<ElementType_x,
+					std::experimental::extents<ext_x>,
+					Layout_x,
+					std::experimental::default_accessor<ElementType_x>
+				      > x,
+				      std::experimental::mdspan<ElementType_y,
+					std::experimental::extents<ext_y>,
+					Layout_y,
+					std::experimental::default_accessor<ElementType_y>
+				      > y)
+{
+
+  // constraints
+  static_assert(A.rank() == 2);
+  static_assert(x.rank() == 1);
+  static_assert(y.rank() == 1);
+
+  // preconditions
+  if ( A.extent(0) != A.extent(1) ){
+    throw std::runtime_error("KokkosBlas: hermitian_matrix_vector_product: A.extent(0) != A.extent(1) ");
+  }
+  if ( A.extent(1) != x.extent(0) ){
+    throw std::runtime_error("KokkosBlas: hermitian_matrix_vector_product: A.extent(1) != x.extent(0) ");
+  }
+  if ( A.extent(0) != y.extent(0) ){
+    throw std::runtime_error("KokkosBlas: hermitian_matrix_vector_product: A.extent(0) != y.extent(0) ");
+  }
+
+  // for non-complex ELementType_A, A can be treated as symmetric
+  // http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p1673r6.html#hermitian-matrix-vector-product-linalgalgsblas2hemv
+  //
+  // If inout_matrix_t::element_type is complex<RA> for some RA, then ....
+  // Otherwise, the functions will assume that A[j,i] equals A[i,j].
+  //
+  if constexpr (hermitianmatvecimpl::is_complex_v<ElementType_A> == false){
```

Comment:
This will not have the desired results if, e.g., `ElementType_A` is `const std::complex<T>` for some `T`.  I've discussed this in my comments on PR #162, among other places.  One fix would be to use the `mdspan`'s `value_type` alias.

Reply:
done